pkg/workflows/management/post_cluster_upgrade.go

```diff
@@ -28,6 +28,10 @@ func (s *postClusterUpgrade) Run(ctx context.Context, commandContext *task.Comma
 		logger.Info(fmt.Sprintf("management cluster CAPI backup file not found: %v", err))
 	}
 
+	if commandContext.CurrentClusterSpec.Cluster.Spec.RegistryMirrorConfiguration != commandContext.ClusterSpec.Cluster.Spec.RegistryMirrorConfiguration {
+		return &upgradeCuratedPackagesTask{}
```

Comment:
I would prefer if the sequence of tasks wasn't decided during runtime
Could we just always return the upgrade task and then inside thta task skip the logic if the registry mirrors are the same?